src/apps/ipv6/reassemble.lua

```diff
@@ -264,24 +284,27 @@ function Reassembler:handle_fragment(h)
       else
          reassembly.final_start = frag_start
       end
-   elseif frag_size % 8 ~= 0 then
+   elseif bit.band(frag_size, 0x7) ~= 0 then
       -- The size of all non-terminal fragments must be a multiple of 8.
       -- Here we should send "ICMP Parameter Problem, Code 0 to the
       -- source of the fragment, pointing to the Payload Length field of
       -- the fragment packet".
       return self:reassembly_error(entry)
    end
 
-   local max_data_offset = ether_ipv6_header_len + frag_start + frag_size
-   if max_data_offset > ffi.sizeof(reassembly.packet.data) then
-      -- Snabb packets have a maximum size of 10240 bytes.
-      return self:reassembly_error(entry)
+   -- Limit the scope of max_data_offset
+   do
+      local max_data_offset = ether_ipv6_header_len + frag_start + frag_size
+      if max_data_offset > ffi.sizeof(reassembly.packet.data) then
+         -- Snabb packets have a maximum size of 10240 bytes.
+         return self:reassembly_error(entry)
```

Comment:
Maximum size of a packet is defined as a constant in `core/packet.lua:max_payload`.

Reply:
Note that this is from the original code (I only added a `do...end` to limit the scope of locals). This would be a cosmetic change, I suppose?

Reply:
Sorry I didn't realize this wasn't a new change. Rather than cosmetic, the difference is that the size is being computed every time this function is called, when it can be retrieved from a constant. I leave it up to you if you'd like to change it.

Reply:
I'd have expected that this would be taken care of by the optimizer, but if it's not, that change would certainly be a good idea.